partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -991,63 +1010,296 @@ class PartiQLSchemaInferencerTests {
-                expected = StaticType.STRING
+                expected = STRING
             ),
             SuccessTestCase(
                 name = "Current User Concat",
                 query = "CURRENT_USER || 'hello'",
-                expected = unionOf(STRING, StaticType.NULL)
+                expected = unionOf(STRING, NULL)
             ),
             SuccessTestCase(
                 name = "Current User Concat in WHERE",
                 query = "SELECT VALUE a FROM [ 0 ] AS a WHERE CURRENT_USER = 'hello'",
-                expected = BagType(StaticType.INT)
+                expected = BagType(INT)
             ),
             SuccessTestCase(
                 name = "TRIM_2",
                 query = "trim(' ' FROM ' Hello, World! ')",
-                expected = StaticType.STRING
+                expected = STRING
             ),
             SuccessTestCase(
                 name = "TRIM_1",
                 query = "trim(' Hello, World! ')",
-                expected = StaticType.STRING
+                expected = STRING
             ),
             SuccessTestCase(
                 name = "TRIM_3",
                 query = "trim(LEADING ' ' FROM ' Hello, World! ')",
-                expected = StaticType.STRING
+                expected = STRING
+            ),
+            SuccessTestCase(
+                name = "Subquery coercion in top-level expression",
+                query = "COALESCE(SELECT identifier FROM aws.ddb.b)",
+                expected = STRING
+            ),
+            SuccessTestCase(
+                name = "Subquery coercion in WHERE. Also showcases conflicting bindings. INT vs STRING.",
+                query = """
+                    SELECT VALUE identifier
+                    FROM aws.b.b AS b -- aws.b.b.identifier is an INT
+                    WHERE
+                        COALESCE(SELECT identifier AS identifier FROM aws.ddb.b) IS NOT NULL -- aws.ddb.b.identifier is a STRING
+                """,
+                expected = BagType(INT)
+            ),
+            SuccessTestCase(
+                name = "Subquery coercion in SFW",
+                query = """
+                    SELECT
+                        (SELECT identifier FROM aws.ddb.b) AS some_str,                         -- identifier is STRING
+                        ('hello' || (SELECT identifier FROM aws.ddb.b)) AS concat_str,          -- identifier is STRING
+                        (1 < (SELECT id FROM aws.ddb.pets)) AS one_lt_id,                       -- id is INT
+                        ((SELECT id FROM aws.ddb.pets) > 1) AS id_gt_one,                       -- id is INT
+                        (1 IN (SELECT id FROM aws.ddb.pets)) AS one_in_ids,                     -- id is INT
+                        ([1, 2] IN (SELECT id, id + 1 FROM aws.ddb.pets)) AS array_in_ids,      -- id is INT
+                        ([1, 2] <= (SELECT id, id + 1 FROM aws.ddb.pets)) AS lit_array_lte_ids  -- id is INT
```

Comment:
nice set of tests through this file. could also test `>=`. don't think `>=` was tested elsewhere

Reply:
ACK. Will add.